XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java

```diff
@@ -58,34 +67,64 @@ public void delete(Claim model) throws RepositoryException {
     }
 
     @Override
-    public void insert(Claim model) throws RepositoryException {
-        throw new UnsupportedOperationException();
-    }
+    public void insert(Claim c) throws RepositoryException {
+        Object[] splitClaim = new Object[5];
+
+        try {
+            if (c == null) {
+                throw new IllegalArgumentException("The object c is null.");
+            }
+            for (Field field : c.getClass().getDeclaredFields()) {
+                field.setAccessible(true);
+                Object value = field.get(c);
+                if (value == null) {
+                    throw new IllegalArgumentException("One of the declared fields in object c is null.");
```

Comment:
This is a cool but flawed trick. I saw this in InsertPaymentService and the problem is this catches the ID field too which is always null - because how are you going to know the ID until insert time?

You'll have to check each field manually

Reply:
The ID is automatically generated remember - the claim model object has no ID, so one isn't passed to the ClaimsRepo method. It has a MEM_ID, but that isn't the primary key, and is determined by the user name!

Reply:
Very true!